app/controllers/withdraw_controller.rb

```diff
@@ -0,0 +1,56 @@
+# frozen_string_literal: true
+
+class WithdrawController < ApplicationController
+  def create
+    @document = Document.with_current_edition.find_by_param(params[:id])
+    edition = @document.current_edition
+    @public_explanation = edition.status&.details&.public_explanation || params[:public_explanation]
+    if !current_user.has_permission?(User::PRE_RELEASE_FEATURES_PERMISSION)
+      render :withdraw
+      return
+    end
+
+    if current_user.has_permission?(User::MANAGING_EDITOR_PERMISSION)
+      render :create
+    else
+      render :non_managing_editor
+    end
+  end
+
+  def new
+    render status: :forbidden if !user_has_permissions?
```

Comment:
I'm not sure what this is rendering? is it the new view? 

It might be more consistent with our app to have an exception with a fixme comment. See - editions and review controllers for example.

Reply:
Now raising an exception if user does not have permissions.